### ISSUE TITLE
Glossary: definition of "prerecorded"

### ISSUE THREAD

**Issue by jkshapiro** [w3c/wcag#323]
The glossary term for "prerecorded" in WCAG2.1, https://www.w3.org/TR/WCAG21/#dfn-prerecorded, simply says "information that is not live". However, the definition for "live", https://www.w3.org/TR/WCAG21/#dfn-live, specifically excludes information that is "completely computer generated".

I think there is computer generated information that doesn't meet the usual English definition of "prerecorded". some examples:
* a real-time clock display
* a screensaver
* a pseudorandom, time-sensitive code used for two-factor authentication

Would it make sense to create a third category for computer generated information? Or, alternatively, to specifically call it out as being considered "prerecorded"?

**Comment by jkshapiro** [comment]
There's also some overlap with https://www.w3.org/TR/WCAG21/#dfn-real-time-events

**Comment by cstrobbe** [comment]
Here are all the phrases from success criteria that use the adjective "prerecorded":

* prerecorded audio-only [media] (SC 1.2.1),
* prerecorded video-only media (SC 1.2.1),
* prerecorded audio content in synchronized media (SC 1.2.2, ,
* prerecorded video content (SC 1.2.3),
* prerecorded audio content in synchronized media (SC 1.2.6, 1.2.7).

With regard to the examples:
1. A real-time clock fits neither the current definition of "precorded" nor the definition of live but is more likely to be provided as updating text than in a video format. (If we interpret video as a sequence of images, a real-time clock in video format fails SC 1.4.5: Images of text.)
2. A screensaver will either disappear or be overlayed with a log-in screen when you do something with an input device. Since you don't interact with the screensaver as such, and it is supposed to run when you're not using your device, I don't see an issue with a screensaver not having subtitles or an audio description.
3. A pseudorandom, time-sensitive code used for two-factor authentication: would this be provided as a video or as audio-only content?

Are there any other types of computer-generated information that would be provided in audio or video format and that would require subtitles or audio description?

**Comment by bruce-usab** [comment]
Are there any other types of computer-generated information that would be provided in audio or video format and that would require subtitles or audio description?

Yes, anything interactive like games or immersive environments.  Those settings need not be "live" but are scripted and programmed and not "prerecorded".

**Comment by patrickhlauke** [comment]
need to be careful not to start overlapping with things that would fall under 1.1.1 Non-text Content or 4.1.3 Status Messages, for instance.

**Comment by GreggVan** [comment]
Yea

We have a new problem with emerging games.   Where the games make up the ’scene’ on the fly.  And they do it with deep learning - -which means they may make up a scene without knowing what the scene is.   That is - they make it look like a room  - -without knowing what the objects in a room are — they just know that rooms look like that and have those kind of things in them.   

So it is KIND OF live — in that it generated the scene on the fly (live) and is not programmed or scripted…  So  Uffda 

This will be one of the topics in the meeting this fall we are doing on  the  Future of Interface  and  Future of Interface Accessibility 
More about that later but wanted to toss awareness of this new wrinkle into the discussion 


Gregg Vanderheiden
***@***.***
